smolyay/basis.py

```diff
@@ -304,24 +344,30 @@ def _derivative(self, x):
         ------
         ValueError
             if input is outside the domain [-1, 1].
-        """
-        x = numpy.asarray(x)
-        y = numpy.zeros(x.shape)
-        u_limit = self.degree * (self.degree + 1) * (self.degree + 2) / 3
-        flag_upper = x == 1
-        y[flag_upper] = u_limit
 
-        flag_lower = x == -1
-        y[flag_lower] = (-1) ** (self.degree + 1) * u_limit
-
-        flag = ~(flag_upper | flag_lower)
-        y[flag] = (
-            (self.degree + 1) * scipy.special.eval_chebyt(self.degree + 1, x[flag])
-            - x[flag] * scipy.special.eval_chebyu(self.degree, x[flag])
-        ) / (x[flag] ** 2 - 1)
-        if y.ndim == 0:
-            y = y.item()
-        return y
+        NotImplementedError
+            Order of derivative outside supported range (1).
+        """
+        if n == 1:
```

Comment:
Rather than need to indent all the code, you can just put this at the top:
```suggestion
        if n != 1:
            raise NotImplementedError("Only first derivative is supported.")
```